db/dcrpg/queries.go

```diff
@@ -1351,19 +1350,34 @@ func InsertVouts(db *sql.DB, dbVouts []*dbtypes.Vout, checked bool) ([]uint64, [
 	return ids, addressRows, dbtx.Commit()
 }
 
-func InsertAddressOut(db *sql.DB, dbA *dbtypes.AddressRow, dupCheck bool) (uint64, error) {
+func RecieveAddressByTxHash(db *sql.DB, tx_hash string) (*dbtypes.AddressRow, error) {
+	var data dbtypes.AddressRow
+	err := db.QueryRow(internal.SelectAddressByTxHash, tx_hash).Scan(&data.InOutRowID, &data.Address, &data.Value)
```

Comment:
Sure we need discuss on this. I thought that every tx_hash is unique??

Reply:
`SelectAddressByTxHash` is a query in to the `addresses` table, which indexes occurrences of addresses in tx inputs and outputs.  Of of course transactions may have numerous inputs and outputs, so in the following query,

```go
SelectAddressByTxHash = `select id, address, value from addresses 
    where tx_hash=$1 and is_funding = true`
```

This loads all inputs and outputs for the transaction with that hash.

Reply:
but `is_funding = true` make a small difference. Question: by default do we have unique `tx_hash` value or the get repeated once in a while?

Reply:
On the blockchain, you will only have 1 valid transaction with a given tx_hash.  However, there may be multiple occurrences of the same transaction as a result of blocks being invalidated by stake votes.  A regular tx that was invalidated will normally be picked up and mined in a different block, but the chain is unbroken so that tx will appear in both places, just only one is valid.

Reply:
But in the addresses table, a specific `tx_hash` value may be repeated many times for the reasons I mentioned.

Reply:
I think here, picking the details of the row with the earliest block time for multiple rows case scenario in the result makes more sense to me here.

Reply:
My current solution.
```
SelectAddressByTxHash = select id, address, value from addresses 
	   where tx_hash=$1 and is_funding = true ORDER BY block_time ASC LIMIT 1;
```